### PR TITLE
[refactor] Switch to the EXPath HTTP Client

### PR DESCRIPTION
**Pull request by adamretter**
We should stop using the legacy HTTP Client as it is not reliable, and our EXPath one now offers much better performance.

This also depends on this issue being fixed - https://github.com/eXist-db/public-xar-repo/issues/17

### CODE REVIEW

**Comment by windauer** [comment]
+1 for not using it in our code! @line-o, @tasmo and me experienced issues with self signed https certs and the public repo (same as with FOP)  but we did not have the capacities to fix this on our own yet. And talking about "outdated", "not reliable" I'd love to get rid of the dashboard and betterform as default. On my fork I'm happily running the new packagemanger from @JoernT and removed dashboard and betterform (yeah I know.. not topic of this issue.. ;) )

File: plugins/packageManager/packages.xql [hunk 1]
```diff
@@ -244,15 +243,7 @@ declare %private function packages:display($repoURL as xs:anyURI?, $app as eleme
...
-                                        { 
-                                            for $change in $app/changelog/change[@version eq $available or packages:is-newer(./@version, $installed)]
-                                            order by $change/@version descending
-                                            return
-                                                <div class="shortTitle">
-                                                    <h3>{$change/@version/string()}</h3>
-                                                    { $change/node() }
-                                                </div>
-                                        }
+                                        { $app/changelog/change[@version = $available]/node() }
```
Comment (joewiz) [inline plugins/packageManager/packages.xql]: This change appears to be out of scope of the PR.

File: plugins/packageManager/packages.xql [hunk 2]
```diff
@@ -122,13 +122,12 @@ declare %private function packages:public-repo-contents($installed as element(ap
...
-(:            </http:request>:)
-(:        let $data := http:send-request($request):)
-        let $data := httpclient:get($url, false(), ())
-        let $status := xs:int($data/@statusCode)
+        let $request :=
+            <http:request method="get" href="{$url}" timeout="10">
+                <http:header name="Cache-Control" value="no-cache"/>
+            </http:request>
+        let $data := http:send-request($request)
+        let $status := xs:int($data[1]//http:response/@statusCode)
```
Comment (joewiz) [inline plugins/packageManager/packages.xql]: The response status is returned in a `@status` attribute, not `@statusCode`. See http://expath.org/spec/http-client#d2e491.

Comment (joewiz) [reply plugins/packageManager/packages.xql]: Also, since `http:send-request()` returns `<http:response>` as the first item, so `$data[1]/@status` should be sufficient as a replacement for `$data[1]//http:response/@status`.

Comment (adamretter) [reply plugins/packageManager/packages.xql]: @joewiz Indeed. I have fixed that now. I couldn't test it until the public-xar-repo issue I mentioned above is resolved.

File: plugins/packageManager/packages.xql (hunk 1 above)
Comment (adamretter) [reply plugins/packageManager/packages.xql]: Oops, I tried to exclude that. I have now fixed that.

Comment (joewiz) [reply plugins/packageManager/packages.xql]: Thanks!

File: plugins/packageManager/packages.xql (hunk 2 above)
Comment (joewiz) [reply plugins/packageManager/packages.xql]: Super, thanks so much.

**Comment by adamretter** [comment]
@JoernT Indeed it does

**Comment by adamretter** [comment]
So this one is approved, but could we test and merge?

**Comment by dizzzz** [comment]
any update on this @joewiz ?

**Comment by joewiz** [comment]
@dizzzz Thanks for the reminder. I just successfully tested the installation of dashboard. I wiped my database and placing the .xar into the autodeploy directory. I was also able to successfully install an app from the public repo with this PR installed. We're good to merge.

I should note that testing was a little tricky, since I usually install apps via the Package Manager, which itself is part of the Dashboard. Installing Dashboard this way would cause the app to overwrite itself. No wonder, then, that Dashboard raised this error while I was trying to install this branch over the existing installation:

> Failed to unpack archive: If the name of an element in an element constructor is in no namespace, creating a default namespace for that element using a computed namespace constructor is an error. Cannot output a namespace node for the default namespace when the element is in no namespace.

Initially I was quite confused by this error message.  Then it occurred to me: perhaps the actual cause is that that Dashboard can't overwrite itself. Unfortunately exist.log didn't provide any other hints about this error.

I confirmed that this error isn't unique to Adam's branch. I wiped the database again and installed the current release of the Dashboard .xar over itself. I got the same error as above. Strangely enough, the request to http://localhost:8080/exist/apps/dashboard/modules/install.xql returns a 200 status.

All to say this PR is in good shape and good to merge. Thanks @adamretter!
